src/aquarium.py

```diff
@@ -81,19 +89,99 @@ async def aquarium_startup(_: FastAPI, aquarium_api: FastAPI):
     )
     gstate.add_storage(storage)
 
+    gstate.init()
+
+
+_shutting_down: bool = False
+_main_task: Optional[asyncio.Task] = None
+
+
+async def aquarium_main_task(
+    app: FastAPI,
+    config: Config,
+    kvstore: KV,
+    gstate: GlobalState,
+    nodemgr: NodeMgr,
+    deployment: DeploymentMgr,
+) -> None:
+    logger.debug("Starting main Aquarium task.")
+
+    app.state.deployment = deployment
+
+    while not _shutting_down and not deployment.installed:
+        logger.debug("Waiting for node to be installed.")
+        await asyncio.sleep(1.0)
+
+    if _shutting_down:
+        return
+
+    assert deployment.installed
+
+    try:
+        await deployment.init()
+    except InitError as e:
+        logger.error(f"Unable to init node: {e.message}")
+        sys.exit(1)
+
+    logger.info("Init Node Manager.")
+    config.init()
+    kvstore.init()
+    gstate_init(gstate, nodemgr)
+    nodemgr.init()
+
+    logger.info("Starting Node Manager.")
     await nodemgr.start()
     await gstate.start()
 
-    # Add instances into FastAPI's state:
-    aquarium_api.state.gstate = gstate
-    aquarium_api.state.nodemgr = nodemgr
+    logger.info("Post-Init Deployment.")
+    deployment.postinit(gstate, nodemgr)
+
+    app.state.gstate = gstate
+    app.state.nodemgr = nodemgr
```

Comment:
This line needs to be moved to line 110 right after `app.state.deployment = deployment` and BEFORE the `while not _shutting_down and not deployment.installed:` loop.